manticore/manticore.py

```diff
@@ -169,8 +170,11 @@ class Manticore(object):
     :ivar dict context: Global context for arbitrary data storage
     '''
 
-    def __init__(self, path_or_state, workspace_url=None, policy='random', **kwargs):
-        
+    def __init__(self, path_or_state, linux_argv=None, linux_envp=None, workspace_url=None, policy='random', **kwargs):
+
+        linux_argv = [] if linux_argv is None else linux_argv
```

Comment:
Why not move this to kwargs  and let Linux() do the check

Reply:
I like keeping the actual arguments, because then it is easy to tell what arguments this function takes. but yes, omit the checks here, and save them for later

Reply:
But we don't really know what arguments this may take as other platforms are be added. I say.. kwargs to the rescue. Like the cooperative classes thing. no?

Reply:
the thought is when platforms are added, they should use their respective classmethod, not the Manticore ctor -- this is a legacy interface only for linux?

Reply:
btw, kwargs is still here, and forwarded to make_initial_state

Reply:
another thought is that we should just break the API - it's mildly inconvenient for some people, but perhaps we are too young to care about legacy cruft?

Reply:
break the API in what way? just in the prototype/how it's documented, but keep old code working?

Reply:
> break the API in what way?

no, i was thinking of _actually_ breaking it. remove how the `Manticore.linux()` interface is currently superimposed over the `Manticore` ctor. break `Manticore('binary', args, envp)` and force use of platform classmethods (unless you already have a State) (`Manticore.linux(binary, args, envp)`)